### PR TITLE
Thirst/hunger update, v2.

### PR DESCRIPTION
**Pull request by EgorDinamit**
## About The Pull Request

- All drinks and water now replenish thirst instantly upon ingestion. This depends on how many units you took.
- Hunger(0.2 -> 0.1) and thirst(0.3 - 0.15) now have lower factors.
- Mixing rad-x and normal water purifies it.
- Changed recipe(Cryptobiolin, Sterilizine, Chlorine) for water purification reagent.
- You can now find water purifier pills in random medical loot.
- More reagents replenish thirst now.

## Why It's Good For The Game

- It's not impossible to replenish thirst now and the change is seen faster, instead of it being on metabolize proc.
- It takes more time to get thirsty/hungry.
- Easier for wastelanders to drink pure water.
- A slightly more difficult recipe so as to not trivialize the process.
- Pills scattered across the wasteland will provide a tiny opportunity to drink pure water without sacrificing rad-x.
- More reagents can help you out with thirst.

### CODE REVIEW

**Comment by BadAtThisGame302** [comment]
I suggest making thirst 0.2.

**Comment by EgorDinamit** [comment]
> I suggest making thirst 0.2.

People can't handle it just yet, that's what we learned today, so let's make it as casual as possible until people learn how to play with it.

**Comment by EgorDinamit** [comment]
Turning this PR into updated version of that system, since it got reverted recently.
**### Do NOT merge for now.**

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -248,8 +248,8 @@
 	taste_description = "water"
 	overdose_threshold = 150 //Imagine drinking a gallon of water
 	var/cooling_temperature = 2
-	var/radiation_amount = 0.5
-	var/thirst_factor = THIRST_FACTOR * 3
+	var/radiation_amount = 0.4
```
Comment (panzerr1944) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:251]: you are STILL giving base ss13 water radiation and causing issues.

**Comment by EgorDinamit** [comment]
Okay, now it's less disgusting.

**Comment by Cartlord** [comment]
We have thirst?

**Comment by EgorDinamit** [comment]
> We have thirst?

It was tested for a round before: #496 
This PR is an updated version after all the feedback from the test merge.

**Pull request merged**